src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -55,10 +55,10 @@ internal sealed partial class ReflectTypeDescriptionProvider : TypeDescriptionPr
         // in.  The keys to the property and event caches are types.
         // The keys to the attribute cache are either MemberInfos or types.
         //
-        private static volatile Hashtable s_propertyCache;
-        private static volatile Hashtable s_eventCache;
-        private static volatile Hashtable s_attributeCache;
-        private static volatile Hashtable s_extendedPropertyCache;
+        private static readonly Lazy<Hashtable> s_propertyCacheLazy = new Lazy<Hashtable>(() => new Hashtable());
+        private static readonly Lazy<Hashtable> s_eventCacheLazy = new Lazy<Hashtable>(() => new Hashtable());
+        private static readonly Lazy<Hashtable> s_attributeCacheLazy = new Lazy<Hashtable>(() => new Hashtable());
+        private static readonly Lazy<Hashtable> s_extendedPropertyCacheLazy = new Lazy<Hashtable>(() => new Hashtable());
```

Comment:
> Just to make sure I understand correctly, LazyInitializer.EnsureInitialized could be preferred over Lazy<T> in all the changes that I am / will be doing?

At least all of the ones where the object in question is a Hashtable.  There's one that's involving a RuntimeBinder, for example, and comments in the ctor for RuntimeBinder suggests it's very expensive to create one, so that might be better as a ```Lazy<T>```.  The issue is that constructing a ```Lazy<T>``` itself has a bit of cost, e.g. it's at least an allocation, so we have to weigh that against no allocation but an optimistic concurrency approach where in the worst case you could have a bunch of threads all racing to initialize and each end up allocating the very expensive thing.